### PR TITLE
Added new option to change default cmk config file.

### PR DESCRIPTION
**Pull request by soreana**
Cloudmonkey command lets us use a different config file rather than the default one. This pull request added the same functionality to the `cmk` command.

```
cmk -c ./my-config
```

### CODE REVIEW

**Review by weizhouapache — APPROVED** [review]

**Comment by weizhouapache** [comment]
tested ok

File: config/config.go
Comment (ravening) [inline config/config.go:333]: What if the file doesnt exist?

Comment (ravening) [reply config/config.go:333]: Its creating config file if the path provided doesnt exist which is not good

```
# ls
bin  CHANGES.md  cli  cmd  cmk.go  config  Dockerfile  go.mod  go.sum  LICENSE  Makefile  performrelease.sh  README.md  snap  vendor
root@mgt01:~/cloudstack-cloudmonkey#
root@mgt01:~/cloudstack-cloudmonkey# ./bin/cmk -c fkTheWholeWorld
Apache CloudStack 🐵 CloudMonkey 6.1.0
Report issues: https://github.com/apache/cloudstack-cloudmonkey/issues

(localcloud) 🐱 > exit
root@mgt01:~/cloudstack-cloudmonkey# ./bin/cmk -c screwCorona
Apache CloudStack 🐵 CloudMonkey 6.1.0
Report issues: https://github.com/apache/cloudstack-cloudmonkey/issues

(localcloud) 🐱 > exit
root@mgt01:~/cloudstack-cloudmonkey# ./bin/cmk -c breakAllStuff
Apache CloudStack 🐵 CloudMonkey 6.1.0
Report issues: https://github.com/apache/cloudstack-cloudmonkey/issues

(localcloud) 🐱 > exit

# ls -lrt
total 96
-rw-r--r-- 1 root root  3926 Oct 29 09:19 README.md
-rw-r--r-- 1 root root  6501 Oct 29 09:19 Makefile
-rw-r--r-- 1 root root 10225 Oct 29 09:19 LICENSE
-rw-r--r-- 1 root root  1498 Oct 29 09:19 Dockerfile
drwxr-xr-x 2 root root  4096 Oct 29 09:19 cli
-rw-r--r-- 1 root root  5294 Oct 29 09:19 CHANGES.md
drwxr-xr-x 2 root root  4096 Oct 29 09:19 cmd
drwxr-xr-x 2 root root  4096 Oct 29 09:19 snap
-rwxr-xr-x 1 root root  5322 Oct 29 09:19 performrelease.sh
-rw-r--r-- 1 root root  4149 Oct 29 09:19 go.sum
-rw-r--r-- 1 root root  1966 Oct 29 09:19 go.mod
drwxr-xr-x 5 root root  4096 Oct 29 09:19 vendor
drwxr-xr-x 2 root root  4096 Oct 29 09:21 config
-rw-r--r-- 1 root root  2108 Oct 29 09:21 cmk.go
drwxr-xr-x 2 root root  4096 Oct 29 09:23 bin
-rw-r--r-- 1 root root   253 Oct 29 09:29 fkTheWholeWorld <<<<<<<<<<<<<<<<<<<
-rw-r--r-- 1 root root   253 Oct 29 09:30 screwCorona <<<<<<<<<<<<<<<<<<<<<<
-rw-r--r-- 1 root root   253 Oct 29 09:30 breakAllStuff. <<<<<<<<<<<<<<<<<<<<<<
```

**Comment by weizhouapache** [comment]
@rhtyd can you look at this pr ?
it works in my testing.

File: config/config.go
Comment (soreana) [reply config/config.go:333]: Thanks @ravening fixed it.

**Comment by rhtyd** [comment]
Need some manual testing, otherwise LGTM

**Comment by rhtyd** [comment]
@weizhouapache there was a new change since your last comment can you and/or @ravening help with manual testing, happy to merge after that.

**Comment by weizhouapache** [comment]
> @weizhouapache there was a new change since your last comment can you and/or @ravening help with manual testing, happy to merge after that.

@rhtyd good. Rakesh and me will test it.

**Comment by weizhouapache** [comment]
tested ok.

1. if /root/.cmk/config does not exist, and it is passed
```
# bin/cmk -c /root/.cmk/config list accounts filter=id
Config file doesn't exist.
```
2. if /root/.cmk/config does not exist, but not passed.  following command will create a new file
```
# bin/cmk list accounts filter=id
🙈 Error: failed to authenticate with the CloudStack server, please check the settings: Post "http://localhost:8080/client/api": dial tcp [::1]:8080: connect: connection refused
```
3. use a pre-defined config
```
# bin/cmk -c /root/.cmk/config.testcloud list accounts filter=id
{
  "account": [
    {
      "id": "1648da77-c83e-4941-80eb-bd01028afbfb"
    }
  ],
  "count": 1
}

```
4. use a non-existing config file
```
# bin/cmk -c /root/.cmk/config.nothing list accounts filter=id
Config file doesn't exist.

```

**Comment by weizhouapache** [comment]
@soreana could you add the new flag to cmd/command.go ?

**Comment by soreana** [comment]
@weizhouapache good point. Added!

**Comment by weizhouapache** [comment]
@rhtyd it is tested ok by me

**Comment by rhtyd** [comment]
Thanks @weizhouapache merging based on your remark.

**Pull request merged**
